### PR TITLE
azure: report status back if build failed

### PR DESCRIPTION
**Pull request by dawidd6**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

This should work once `GITHUB_TOKEN` is added as secret in Azure settings.
I believe the token needs only **repo:status** scope, but I can be wrong.

### CODE REVIEW

**Comment by iMichka** [comment]
I added a token, it is called AZURE_BUILD_STATUS. Github told me that GITHUB_TOKEN is not allowed as a token name.

To test if this works: revision bump the hello formula, remove the sha256, and add a small style error (for example replace http by https or add a trailing whitespace): the build status should be failed, but the bottle upload should be green.

**Comment by iMichka** [comment]
I think you need to keep `set -e`.

**Comment by iMichka** [comment]
> cellar: old: :any_skip_relocation, new: "/home/linuxbrew/.linuxbrew/Cellar"

You also need to add that workaround too, as no bottle was built.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -43,15 +43,36 @@ jobs:
...
+        echo "TOKEN = $AZURE_BUILD_STATUS"
+        exit 1
+      fi
     displayName: Run test bot
     env:
       HOMEBREW_DEVELOPER: 1
       HOMEBREW_NO_AUTO_UPDATE: 1
       HOMEBREW_VERBOSE: 1
       HOMEBREW_VERBOSE_USING_DOTS: 1
+      AZURE_BUILD_STATUS: $(AZURE_BUILD_STATUS)
```
Comment (dawidd6) [inline azure-pipelines.yml]: I think this line did the trick.

**Pull request merged**